sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -2844,6 +2844,25 @@ abstract class JsonSuite
       assert(readback.collect sameElements Array(Row(0), Row(1), Row(2)))
     }
   }
+
+  test("Write Non-ASCII character as codepoint") {
+    // scalastyle:off nonascii
+    withTempPath { path =>
+      val basePath = path.getCanonicalPath
+      Seq("a", "\n", "\u3042").toDF.write
+        .option("writeNonAsciiCharacterAsCodePoint", "true").json(s"$basePath")
+      val actualText = spark.read.text(s"$basePath")
+        .sort("value").map(_.getString(0)).collect().mkString
+      val expectedText = "{\"value\":\"\\n\"}{\"value\":\"\\u3042\"}{\"value\":\"a\"}"
+      assert(actualText === expectedText)
+
+      val actualJson = spark.read.json(s"$basePath")
+        .sort("value").map(_.getString(0)).collect().mkString
+      val expectedJson = "\na\u3042"
+      assert(actualJson === expectedJson)
+    }
+    // scalastyle:on nonascii
+  }
```

Comment:
It would be great if we can have a test coverage for setting both pretty and non-ascii-as-codepoint at the same time.

Reply:
O.K, I'll cover the case of `pretty=true`.
BTW, I've noticed another issue when writing the test case for `pretty=true` (#32203).
So, I'll fix that issue first.

Reply:
#32203 was merged and I've rebased to `master`.